src/Components/Web.JS/src/Services/NavigationUtils.ts

```diff
@@ -47,6 +47,25 @@ export function isWithinBaseUriSpace(href: string) {
   && (nextChar === '' || nextChar === '/' || nextChar === '?' || nextChar === '#');
 }
 
+export function isSamePageWithHash(absoluteHref: string): boolean {
+  const hashIndex = absoluteHref.indexOf('#');
+  return hashIndex > -1 && location.href.replace(location.hash, '') === absoluteHref.substring(0, hashIndex);
+}
```

Comment:
We should stop doing stuff like this and start using new URL(absoluteHref) for performing comparisons against the individual pieces that we care about.

Reply:
Yeah the `.replace` technique is really weird, and quite hard to reason about whether it's really correct. I think it is, but it takes time to think through what happens in cases like "current page is `#abc#abc` and you navigate to `#abc`" - it does work but only because of the weirdness that JS's string.replace only replaces the first match, not all matches.